source/client/RunManagement.py

```diff
@@ -0,0 +1,250 @@
+wilds and Aces on ends of runs the appropriate tempnumber before processRuns is called.
+To preserve info on whether Ace is assigned hi or low, if Ace is assigned low, then tempnumber is set to -1.
+"""
+
+def processRuns(card_group, wild_numbers):
+    """ handle sorting of run, including placement of wilds.  Handles minor rule checking.
+
+    # processRuns does not presume length requirement or that all cards are in same suit.
+    # it DOES presume that if Aces are not wild, then they are hi or low, but not both.
+    # IF ACE CAN BE HIGH OR LOW (very unusual) THAN AUTOMATICALLY MAKING IT LOW.
+    """
+    # print("in RunManagement, processRuns")
+    # print(card_group)
+    card_group.sort(key=lambda wc: wc.tempnumber)
+    # print(card_group)
+    first_card = True
+    groups_wilds = []
+    temp_run_group = []
+    aces_list =[]
+    gap_flag = False
+    for card in card_group:                    # separate unassigned wilds and Aces from rest of run.
+        # pull out wilds that have not been assigned (do this before Aces, in case rules changed so that Aces are wild).
+        if card.tempnumber in wild_numbers:
+            groups_wilds.append(card)
+            # pull out Aces that have not been assigned hi or low, but don't pull out Wilds assigned to low Ace slot.
+        elif card.tempnumber == 1 and card.number == 1:
+            aces_list.append(card)
+        else:
+            temp_run_group.append(card)
+    card_group  = []                         # rebuild card_group below
+    for card in temp_run_group:
+        if first_card:
+            first_card = False
+            card_group.append(card)
+        else:
+            abs_temp_number = abs(card_group[-1].tempnumber)  # Aces designated to be low have tempnumber = -1.
+            if card.tempnumber == (abs_temp_number + 1):
+                card_group.append(card)
+            elif card.tempnumber == (abs_temp_number + 2) and len(groups_wilds) > 0:
+                this_wild = groups_wilds.pop(0)
+                this_wild.tempnumber = abs_temp_number + 1
+                card_group.append(this_wild)
+                card_group.append(card)
+            elif card.tempnumber == abs_temp_number:
+                if isWild(card, wild_numbers):
+                    card.tempnumber = card.number  # reset Wild card back to original value (0 for Jokers)
+                    groups_wilds.append(card)
+                elif isWild(card_group[-1], wild_numbers):
+                    this_wild = card_group.pop(-1)
+                    this_wild.tempnumber = this_wild.number
+                    groups_wilds.append(this_wild)
+                    card_group.append(card)
+                else:
+                    raise Exception('Card value already in the run.')
+            elif card.tempnumber > (abs_temp_number + 2):
+                raise Exception('too big a gap between numbers')
+            else:
```

Comment:
This represents "absValue is lastCard +2 and there is no wild card" which would be less confusing if it were with "absValue is lastCard +2 and there IS a wild card" in an if/else case inside the "absValue is lastCard+2 elif" earlier (remove the and and add nested if/else.
Then the elif about too big a gap becomes the else and you can let the exception explain what case it is.

Reply:
Rearranged else statements so that more clear (they had previously grown a little too organically).  Also did some simplification of Ace section -- removed code that was written when I thought that I was going to write code to select whether Aces should be high or low.